lnrpc/walletrpc/psbt.go

```diff
@@ -56,7 +56,14 @@ func lockInputs(w lnwallet.WalletController,
 			},
 		}
 
-		expiration, pkScript, value, err := w.LeaseOutput(
+		// Get the details about this outpoint.
+		utxo, err := w.FetchInputInfo(&lock.Outpoint)
```

Comment:
I'm not sure this is semantically identical with the previous behavior.
Because I think `LeaseOutput` only requires the output to be a known confirmed or unconfirmed UTXO.
But `FetchInputInfo` has a specific cast to `pubKeyAddr, ok := addr.(waddrmgr.ManagedPubKeyAddress)`, which for example a Taproot address wouldn't pass.
So I think this would break things in `tapd`.

Maybe we should instead just return the value of the "credit" in `wtxmgr.isKnownOutput()` and use that value, return it in `Store.LockOutput`?

Reply:
because the address is not managed in `lnd`'s wallet? Think there's also the `GetTxOut` we can use, but yeah I like your suggestion as it saves us one lookup, will create it in `btcwallet`.

Reply:
> because the address is not managed in lnd's wallet?

It is managed by the wallet. But the root interface is `waddrmgr.ManagedAddress`, not `waddrmgr.ManagedPubKeyAddress`, so the cast would fail and return `ErrNotMine`.

Reply:
ok tried it out locally, we can easily get the output value, but not the `pkScript` unless we deserialize the tx details, which I think can be a large overhead if we just want to lock an input.

> But the root interface is waddrmgr.ManagedAddress

Maybe a dumb question but can we change it to be `waddrmgr.ManagedPubKeyAddress` instead?

I think I'll just add a `ListLeasedOutputs` here instead and perform a lookup using the old way used in `LeaseOutput` - at least for those callsite who don't care about the value and script the speed is saved.

Reply:
> Maybe a dumb question but can we change it to be waddrmgr.ManagedPubKeyAddress instead?

Unfortunately not. A `waddrmgr.ManagedPubKeyAddress` means there's a single public key associated with an address. But a Taproot address can have scripts, so it's a `waddrmgr.ManagedScriptAddress`...

Maybe we can add refactor `FetchInputInfo` into two versions: `FetchInputTxInfo` which does everything _but_ the `*psbt.Bip32Derivation` that can be used here. And then the original `FetchInputInfo` calls into that and then does the cast to `waddrmgr.ManagedPubKeyAddress` to extract the derivation info.

Reply:
Cool implemented as suggested.